ssh/client.go

```diff
@@ -261,9 +260,17 @@ func NewExternalClient(sshBinaryPath, user, host string, port int, auth *Auth) (
 	// the authorized hosts file.
 	args := baseSSHArgs
 
+	// If no identities are explicitly provided, also look at the identities
+	// offered by ssh-agent
+	if len(auth.Keys) > 0 {
+		args = append(args, "-o", "IdentitiesOnly=yes")
+	}
+
 	// Specify which private keys to use to authorize the SSH request.
 	for _, privateKeyPath := range auth.Keys {
```

Comment:
This would work in most cases, but most operating systems (debian, ubuntu, etc) will fail after a few attempts.  I don't think this will affect us as I believe we only add the single key we generate but just to note.